src/components/core/geoChart/geoChartColor.ts

```diff
@@ -73,15 +73,34 @@ export function getPushpinColors(
     colorValues: number[],
     segmentItems: Execution.IResultHeaderItem[] = [],
 ): IPushpinColor[] {
-    if (!colorValues.length) {
+    if (!colorValues.length && !segmentItems.length) {
         return [];
     }
+    const segmentItemNames: string[] = segmentItems.map(
+        (item: Execution.IResultHeaderItem): string => {
+            const name: string = getHeaderItemName(item);
+            return name || EMPTY_SEGMENT_ITEM;
+        },
+    );
```

Comment:
Duplicate with https://github.com/gooddata/gooddata-react-components/pull/1353/files#diff-401c51ce84aba39a6e5b2ba4c990ca86R130, Consider to separate new function

Reply:
getHeaderItemName is a separate function, so it could be:
`return getHeaderItemName(item) ||  EMPTY_SEGMENT_ITEM `